step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/TimeSeriesService.java

```diff
@@ -69,9 +70,17 @@ public TimeSeriesAPIResponse getTimeSeries(FetchBucketsRequest request) {
     @Produces(MediaType.APPLICATION_JSON)
     // TODO this method should be renamed as it doesn't return measurements but a timeseries
     public TimeSeriesAPIResponse getMeasurements(FetchBucketsRequest request) {
+        enrichRequestFilter(request);
         return handler.getOrBuildTimeSeries(request);
     }
 
+    private void enrichRequestFilter(FetchBucketsRequest request) {
+        String additionalOqlFilter = getObjectFilter().getOQLFilter();
+        if (additionalOqlFilter != null && ! additionalOqlFilter.isEmpty()) {
+            request.setOqlFilter(request.getOqlFilter() + " and (" + additionalOqlFilter + ")");
```

Comment:
Not sure if this is really safe. Can request.getOqlFilter be null or empty?

Reply:
there was indeed a risk, updated in last commit